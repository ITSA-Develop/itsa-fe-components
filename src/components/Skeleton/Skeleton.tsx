import { Skeleton as AntSkeleton, SkeletonProps } from 'antd';

export const Skeleton = ({ ...rest }: SkeletonProps) => {
	return <AntSkeleton {...rest} active />;
};
