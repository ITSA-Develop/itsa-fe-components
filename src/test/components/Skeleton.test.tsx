import { render } from '@testing-library/react';
import { Skeleton } from '../../components/Skeleton/Skeleton';

describe('Skeleton component', () => {
	it('renders the Skeleton with active class by default', () => {
		const { container } = render(<Skeleton />);
		const skeletonDiv = container.querySelector('.ant-skeleton');
		expect(skeletonDiv).toBeInTheDocument();
		expect(skeletonDiv).toHaveClass('ant-skeleton-active');
		expect(container).toMatchSnapshot();
	});

	it('passes extra props to AntSkeleton', () => {
		const { container } = render(<Skeleton paragraph={{ rows: 3 }} />);
		const paragraphs = container.querySelectorAll('.ant-skeleton-paragraph > li');
		expect(paragraphs.length).toBe(3);
		expect(container).toMatchSnapshot();
	});
});
