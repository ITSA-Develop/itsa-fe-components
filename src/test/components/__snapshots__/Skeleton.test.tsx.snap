// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton component > passes extra props to AntSkeleton 1`] = `
<div>
  <div
    class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-skeleton-content"
    >
      <h3
        class="ant-skeleton-title"
        style="width: 38%;"
      />
      <ul
        class="ant-skeleton-paragraph"
      >
        <li />
        <li />
        <li
          style="width: 61%;"
        />
      </ul>
    </div>
  </div>
</div>
`;

exports[`Skeleton component > renders the Skeleton with active class by default 1`] = `
<div>
  <div
    class="ant-skeleton ant-skeleton-active css-dev-only-do-not-override-l9pxc0"
  >
    <div
      class="ant-skeleton-content"
    >
      <h3
        class="ant-skeleton-title"
        style="width: 38%;"
      />
      <ul
        class="ant-skeleton-paragraph"
      >
        <li />
        <li />
        <li
          style="width: 61%;"
        />
      </ul>
    </div>
  </div>
</div>
`;
